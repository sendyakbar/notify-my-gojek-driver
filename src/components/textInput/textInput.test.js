import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TextInput from './input.jsx'

describe('<TextInput> component', () => {
    const wrapper = shallow(<TextInput />)
    test('renders correctly', () => {
        expect(wrapper).toHaveLength(1)
    })
    test('it has a div', () => {
        const divWrapper = wrapper.find('div')
        expect(divWrapper).toHaveLength(1)
    })
    test('it has a form', () => {
        const formWrapper = wrapper.find('form')
        expect(formWrapper).toHaveLength(1)
    })
    test('it has an input', () => {
        const inputWrapper = wrapper.find('input')
        expect(inputWrapper).toHaveLength(1)
    })
    test('renders correctly and matched snapshot', () => {
        const tree = renderer.create(<TextInput />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})