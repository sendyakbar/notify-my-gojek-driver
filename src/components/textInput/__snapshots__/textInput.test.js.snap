// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput> component renders correctly and matched snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
    />
  </form>
</div>
`;
